wordpress-mobile/WordPress-Android
# Pages List: Add Copy link functionality for pages

Fixes #15612 

To test:

1. Visit list of your published pages
2. Select the context menu (three dots)
3. Observe "copy link" option
4. Click "copy link" and observe link copied to clipboard
5. Visit list of draft pages
6. Select the context menu (three dots)
7. Observe lack of "copy link" option - Only published pages have a link

## Regression Notes
1. Potential unintended areas of impact
Added a new param to onMenuAction, which is used in multiple areas of the app and quite a few tests. Possible impact in those areas.

4. What I did to test those areas of impact (or what existing automated tests I relied on)
Only copying link needs the context param - Made it nullable and added a default value to avoid modifying existing calls to the function.

6. What automated tests I added (or what prevented me from doing so)
Added:
`when copying page link, it copies to device clipboard`
Modified:
```
verify PUBLISHED actions
verify PUBLISHED actions contain HOMEPAGE settings when site has static homepage and is WPCom
verify PUBLISHED actions contain HOMEPAGE settings when site has statis homepage and is Jetpack
verify PUBLISHED actions cannot set page as homepage when it is already set
verify PUBLISHED actions cannot set page for posts when it is already set
verify PUBLISHED actions does not contant HOMEPAGE settings when site has no remote id
verify PUBLISHED actions does not contant HOMEPAGE settings when site is self hosted
```

PR submission checklist:
